lib/agenda.js

```diff
@@ -609,11 +609,16 @@ Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
 
   var self = this, now = new Date(), lockDeadline = new Date(Date.now().valueOf() - definition.lockLifetime);
 
-  // Don't try and access MongoDB if we've lost connection to it.
-  // Trying to resolve crash on Dev PC when it resumes from sleep. NOTE: Does this still happen?
   var s = this._mdb.s || this._mdb.db.s;
   if (s.topology.connections().length === 0) {
-    cb(new Error('No MongoDB Connection'));
+    if (s.topology.autoReconnect && !s.topology.isDestroyed()) {
+      // continue processing but notify that Agenda has lost the connection
+      self.emit('error', new Error('No MongoDB connection'));
+      cb();
+    } else {
+      // no longer recoverable
+      cb(new Error('MongoDB topology has been destroyed'));
```

Comment:
Since a lot of people aren't going to understand what "topology has been destroyed" maybe we could provide a nicer error and stick that type of warning in the debug statement in https://github.com/agenda/agenda/pull/468?